src/microsim/schema/simulation.py

```diff
@@ -168,6 +186,104 @@ def _truth_cache_path(
+        if len(fluorophores) == 0:
+            # we have no fluorophores to calculate
+            raise ValueError("No fluorophores in the current sample!")
+
+        # Get emission spectra for all the fluorophores
+        fluor_em_spectra = []
+        for fluor in fluorophores:
+            if fluor is None:
+                raise NotImplementedError(
+                    "Fluorophore must be defined, instead we got `None`"
+                )
+            else:
+                # get emission Spectrum for the given fluorophore
+                fluor_em_spectra.append(fluor.emission_spectrum.wavelength.magnitude)
+
+        # Get the min and max wavelength over all the spectra
+        min_wave = min([x.min() for x in fluor_em_spectra])
+        max_wave = max([x.max() for x in fluor_em_spectra])
+
+        # Create the same bins for all the spectra
+        wave_range = np.arange(min_wave, max_wave, 1)
+        em_bins = generate_bins(
+            x=wave_range,
+            y=None,
+            num_bins=self.settings.num_wavelength_bins,
+            strategy=self.settings.binning_strategy,
+        )
+
+        # TODO: is this the right format for returning the bins?
+        return em_bins
+
+    def illumination_flux(
+        self,
+        truth: xr.DataArray | None = None,
+        channel_idx: int = 0,
+        *,
+        light_power: float = 100,
+        min_wavelength: float = 300,
+        max_wavelength: float = 800,
+    ) -> xr.DataArray:
+        """
+        Return the illumination data as an array of shape (W, C, Z, Y, X).
+
+        NOTE: we assume this happens before the excitation filters are applied.
+
+        NOTE: for the moment we assume the light source to be the same over all
+        the spatial dimensions. Only dimension is the wavelength.
+        """
+        if truth is None:
+            truth = self.ground_truth()
+        elif not isinstance(truth, xr.DataArray):
+            raise ValueError("truth must be a DataArray")
+
+        channel = self.channels[channel_idx]
+        illum = channel.illumination
+        if not illum:
+            # If illumination is not defined, we assume a white light source
+            illum = Spectrum(
+                wavelength=np.arange(min_wavelength, max_wavelength, 1),
+                intensity=np.ones(max_wavelength - min_wavelength),
+                scalar=light_power,
+            )
+        # Bin illum spectrum
+        binned_illum = bin_spectrum(
+            spectrum=illum, bins=self.wavelength_bins
+        )  # shape: (W)
+        binned_illum
+        # Broadcast to (W, Z, Y, X)
+        binned_illum = binned_illum.expand_dims(
+            [Axis.Z, Axis.Y, Axis.X], axis=[1, 2, 3]
+        )
+        spatial_illum = binned_illum * np.ones(
+            (1, *truth.shape[1:])
+        )  # shape: (W, Z, Y, X)
+        spatial_illum = spatial_illum.expand_dims(
+            [Axis.C], axis=1
+        )  # shape: (W, 1, Z, Y, X)
```

Comment:
```suggestion
        # Bin illum spectrum, shape: (W)
        binned_illum = bin_spectrum(spectrum=illum, bins=self.wavelength_bins)
        # Broadcast to (W, Z, Y, X)
        binned_illum = binned_illum.expand_dims(
            [Axis.Z, Axis.Y, Axis.X], axis=[1, 2, 3]
        )
        # shape: (W, Z, Y, X)
        spatial_illum = binned_illum * np.ones((1, *truth.shape[1:]))
        # shape: (W, 1, Z, Y, X)
        spatial_illum = spatial_illum.expand_dims([Axis.C], axis=1)
```

bugs me when a comment is the only thing causing a line break :)  in that case, put the comment above the line